site/en/docs/extensions/mv3/service-workers/service-worker-lifecycle/index.md

```diff
@@ -0,0 +1,68 @@
+
+The first event fired during installation is  a web service worker's [install](https://developer.mozilla.org/docs/Web/API/ServiceWorkerGlobalScope/install_event) event.
+
+### chrome.runtime.onInstalled
+
+Next is the extension's `oninstalled` event, which is fired when the extension (not the service worker) is first installed, when the extension is updated to a new version, and when Chrome is updated to a new version. Use this
+event to set a state or for one-time initialization, such as a [context menu](/docs/extensions/reference/contextMenus/).
+
+```js
+chrome.runtime.onInstalled.addListener(() => {
+  chrome.contextMenus.create({
+    "id": "sampleContextMenu",
+    "title": "Sample Context Menu",
+    "contexts": ["selection"]
+  });
+});
+```
+### ServiceWorkerRegistration.active
+
+Finally, the  service worker's [activate](https://developer.mozilla.org/docs/Web/API/ServiceWorkerGlobalScope/activate_event) event is fired. Note that unlike web service workers, this event is fired immediately after installation of an extension because there is nothing comparable to a page reload in an extension.
+
+## Extension startup
+
+When a user or incognito profile starts, the [`chrome.runtime.onStartup`](/docs/extensions/reference/runtime/#event-onStartup) event fires but no service worker events are invoked.
+
+## Idle and shutdown
+
+Extension service workers are dormant unless an event fires. If the service worker has a handler for the fired event, it wakes up, executes it, then goes back to being idle. If the service worker is idle for at least 30 seconds it shuts down. You should design your service worker to be resilient against unexpected termination, since not all activity keeps the service worker alive. For example, this can interrupt a service worker `fetch()` call if the response takes more than 30 seconds to arrive.
+
+Any new events and calls to extension APIs reset the idle timer the moment they're fired. For example, when a service worker receives a `chrome.bookmarks.onCreated` event, the 30 second timeout is reset. The same is true when calling an extension API such as `chrome.storage.local.get()` which also resets the 30 second timeout. As with web service workers, extension service workers have no shutdown or deactivation events.
+
+Before Chrome 110, only running event handlers caused the idle time to reset. Any events that were queued, but for which a handler had not been called would not cause a reset. Also, extension service workers had a maximum lifetime of five minutes before Chrome shut them down. These behaviors caused service workers to shut down at unexpected times.
+
+Though Chrome 110 changed this you should not keep your service worker alive indefinitely, though you can. (We don't consider this a good programming practice.) You should test your extensions to ensure that they're not doing this unintentionally.
+
+Though it's difficult to catch in a code review, we suggest you guard against it as best you can.
+
```

Comment:
Can we add a subheader here like "### Storing data" or "### Persisting data"

Reply:
Line 54 was supposed to have been a heading. It's fixed.